it/test/connectors/ITSAStatusUpdateConnectorISpec.scala

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors
+
+import _root_.helpers.{ComponentSpecBase, WiremockHelper}
+import connectors.itsastatus.ITSAStatusUpdateConnectorModel.{ErrorItem, ITSAStatusUpdateResponseFailure, ITSAStatusUpdateResponseSuccess}
+import connectors.itsastatus.{ITSAStatusUpdateConnector, ITSAStatusUpdateRequest}
+import models.incomeSourceDetails.TaxYear
+import org.scalatest.wordspec.AnyWordSpec
+import play.api.http.Status.{BAD_REQUEST, NO_CONTENT, OK}
+import play.api.libs.json.Json
+
+class ITSAStatusUpdateConnectorISpec extends AnyWordSpec with ComponentSpecBase {
+
+  lazy val connector: ITSAStatusUpdateConnector = app.injector.instanceOf[ITSAStatusUpdateConnector]
+
+  val optOutUpdateReason = "10"
+  val optInUpdateReason = "11"
+  val taxYear2024 = TaxYear.forYearEnd(2024)
+  val taxYear2023 = TaxYear.forYearEnd(2023)
+  val taxableEntityId = "AB123456A"
+
+  val correctOptOutRequestBody =
+    """{
+      | "taxYear":"2023-24",
+      | "updateReason":"10"
+      |}""".stripMargin
+
+  val correctOptIntRequestBody =
+    """{
+      | "taxYear":"2023-24",
+      | "updateReason":"11"
+      |}""".stripMargin
+
+  "ITSAStatusUpdateConnector" when {
+
+    ".updateITSAStatus()" when {
+
+      "happy path" should {
+
+        "return a successful response" in {
+
+          WiremockHelper.stubPut(s"/income-tax-view-change/itsa-status/update/$taxableEntityId", NO_CONTENT, "{}")
```

Comment:
Would it be possible to add the body to the matching with the stubbing? That's one part of the conversion to v2 that could easily break and without this check it's hard to ensure we've not broken the behaviour.

Is it also possible to split the commits to add the ISpec first as one commit? That way it can be seen to run against the old V1 code first?

Reply:
would adding the Ispec on a separate branch work? https://github.com/hmrc/income-tax-view-change-frontend/pull/2553

Reply:
could maybe try adding: `.withRequestBody(equalToJson(expectedRequestBody)) ` to the stub?

```
def stubPut(url: String, status: Integer, expectedRequestBody: String, responseBody: String): StubMapping =
    stubFor(
      put(urlEqualTo(url))
        .withRequestBody(equalToJson(expectedRequestBody)) 
        .willReturn(
          aResponse()
            .withStatus(status)
            .withBody(responseBody)
        )
    )
```

will give it a go.

Reply:
Sure, another branch would definitely work.

And yes, something using `withRequestBody()` should do the trick.

Reply:
done